RNA-seq differential expression analysis
========================================
> To knit .rmd file, read data files in using "../data"  
> To run chunks in Rstudio, read data files in using "./data"

Use `voom + limma` to perform differential expression analysis on the RNA-seq data.

Load required libraries:
```{r warning=FALSE, message=FALSE}
library(RColorBrewer)
library(reshape2)
library(plyr)
library(ggplot2)
library(limma)
library(edgeR)
```

Load RNA-seq data and the experimental design:
```{r}
rDat <- read.table("../data/aml.rnaseq.gaf2.0_rpkm_cleaned.txt", sep = "\t",
                   header = TRUE, check.names = FALSE)
rDes <- read.delim("../data/experimental_design_cleaned.txt")
```

Data inspection:
```{r}
str(rDat, max.level = 0)
str(rDes, max.level = 0)
```

```{r include=FALSE}
# More data inspection
rDat[1:4, 1:4]
head(names(rDat))
head(rownames(rDat), n = 10)
tail(rownames(rDat), n = 10)
```


## Differential expression analysis
I will use `voom` to perform differential expression analysis. From my experience, `voom` makes the most stringent calls for differential expression. 

**Sex**  
First I will do a simple differential expression analysis: which genes are differentially expressed between males and females?
```{r}
sex <- rDes$Sex
table(sex)
```

Apply scale normalization. I'm not sure if I need to do this given we have RPKM values?
```{r}
normFactor <- calcNormFactors(rDat)
```

Linear modelling: Use `voom` to convert RPKM to log2-RPKM ready for linear modelling:
```{r voomMeanVarTrendPlot_rpkm}
design <- model.matrix(~ sex)
# The intercept represents females
head(design)
rDatVoom <- voom(rDat, design, plot = TRUE)
```

Now find genes differentially expressed between males and females:
```{r}
fit <- lmFit(rDatVoom, design)
fit <- eBayes(fit)
voomSex <- topTable(fit, coef = "sexM", p.value = 1e-5, n = Inf)
nrow(voomSex)
head(voomSex, n = 10)
(voomSexgenes <- rownames(voomSex))
# Lower the FDR threshold: how many genes do we find?
nrow(topTable(fit, coef = "sexM", p.value = 1e-2, n = Inf))
```
Ok XIST and TSIX are popping up, this is a promising result! Plus it turns out the top hit, `r voomSexgenes[1]` is "Ribosomal Protein S4, Y-Linked 1", so I must be doing something right :)

Can I plot the differentially expressed genes using a DGEList object and plotSmear function from `edgeR`?
```{r plotSmear_Sex_rpkm}
# Create a DGEList object
dgeGlm <- DGEList(counts = rDat, group = sex)
plotSmear(dgeGlm, de.tags = voomSexgenes, ylab = "logFC", xlab = "AverageRPKM")
abline(h = c(-1, 1), col = "blue")
```
Why aren't all genes with > 1 and < -1 logFC not called as differentially expressed?

I want to check the expression of the genes in the smear plot that had logFC < -1 between males and females yet were not called as differentially expressed:
```{r}
voomSexFCtest <- topTable(fit, coef = "sexM", n = Inf)
voomSexFCtest$transcript <- rownames(voomSexFCtest)
voomSexFCtest <- arrange(voomSexFCtest, logFC)
nrow(voomSexFCtest)
voomSexFCtest <- subset(voomSexFCtest, logFC < -1)
nrow(voomSexFCtest)
voomSexFCtest
```
Ok this is really weird, why can't I find the genes on the smear plot? There should be 6 genes in total with logFC < -1 yet I can only find the 2 that were called as differentially expressed??? I'm not sure that I understand what the smear plot is showing anymore, given I can't find transcripts that meet a certain criteria given the results of this plot...

Let's plot the top hits from differential expression analysis to see if I'm on the right track:
```{r sex_MalevsFemale_TopSixBoxplot_rpkm, tidy=FALSE}
rDatvoomSex <- rDat[voomSexgenes[1:6], ]
dim(rDatvoomSex)
rDatvoomSex$Transcript <- rownames(rDatvoomSex)
rDatvoomSex <- melt(rDatvoomSex, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "RPKM")
rDatvoomSex$Transcript <- gsub("[|].*$", "", rDatvoomSex$Transcript)
head(rDatvoomSex)
rDatvoomSex <- merge(rDatvoomSex, rDes, by = "TCGA_patient_id")
head(rDatvoomSex)
ggplot(rDatvoomSex, aes(Transcript, RPKM, colour = Sex)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free")+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
```
So the top genes differentially expressed between males and females have almost no expression in one of the sexes. Given we only found `r nrow(voomSex)` genes differentially expressed with an FDR of 1e-5, we can conclude that sex has limited influence on gene expression in AML patients.


**Cytogenetic risk**  
Now to explore another variable, "Cytogenetic_risk". 
```{r}
levels(rDes$Cytogenetic_risk)
table(rDes$Cytogenetic_risk)
```

Which transcripts are differentially expressed between "Good", "Intermediate", and "Poor" cytogenetic risk?

First, remove samples where cytogenetic risk could not be determined "N.D":
```{r}
# CRGIP = Cytogenetic Response Good + Intermediate + Poor
rDesCRGIP <- droplevels(subset(rDes, Cytogenetic_risk != "N.D."))
str(rDesCRGIP)
dim(rDesCRGIP)
rDatCRGIP <- rDat[ , names(rDat) %in% rDesCRGIP$TCGA_patient_id]
dim(rDatCRGIP)
identical(names(rDatCRGIP), as.character(rDesCRGIP$TCGA_patient_id))
cytoRisk <- rDesCRGIP$Cytogenetic_risk
```

Now I will make a model with a reference + treatment effect, where the intercept is the reference:
```{r rDatCRGIP_voomMeanVarTrendPlot_rpkm}
normFactor <- calcNormFactors(rDatCRGIP)
design <- model.matrix(~ cytoRisk)
colnames(design)
# The intercept represents "Good" cytogenetic risk
head(design)
rDatCRGIPvoom <- voom(rDatCRGIP, design, plot = TRUE)
fit <- lmFit(rDatCRGIPvoom, design)
fit <- eBayes(fit)
voomCR <- topTable(fit, coef = c("cytoRiskIntermediate", "cytoRiskPoor"), 
                   p.value = 1e-5, n = Inf)
nrow(voomCR)
head(voomCR)
voomCRgenes <- rownames(voomCR)
```

```{r include=FALSE}
# dgeGlm <- DGEList(counts = rDatCRGIP, group = cytoRisk)
# plotSmear(dgeGlm, de.tags = voomCRgenes, ylab = "logFC", xlab = "AverageRPKM")
# abline(h = c(-1, 1), col = "blue")
# # Ok this doesn't look right. There are genes coloured red with very little logFC? But hang on, plotSmear plots logCF between two experimental groups. It has chosen to plot Good against intermediate by default???
```

Genes differentially expressed between Good and Poor cytogenetic risk:
```{r}
voomCRGP <- topTable(fit, coef = "cytoRiskPoor", p.value = 1e-5, n = Inf)
nrow(voomCRGP)
voomCRGPgenes <- rownames(voomCRGP)
```

Let's plot the top 6 hits to see if I'm on the right track:
```{r CytoRisk_GoodvsPoor_TopSixBoxplot_rpkm, tidy=FALSE}
rDatvoomCR <- rDatCRGIP[voomCRGPgenes[1:6], ]
dim(rDatvoomCR)
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "RPKM")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, RPKM, colour = Cytogenetic_risk)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())  
```

What about the hits with little logFC? Let's plot these:
```{r}
voomCRGP <- topTable(fit, coef = "cytoRiskPoor", p.value = 1e-5, n = Inf,
                     sort.by = "logFC")
```

```{r include=FALSE}
head(voomCRGP)
tail(voomCRGP)
nrow(voomCRGP)
```

```{r CytoRisk_GoodvsPoor_lowlogFC_Boxplot_rpkm}
voomCRGPgenes <- rownames(voomCRGP)
rDatvoomCR <- rDatCRGIP[tail(voomCRGPgenes), ]
dim(rDatvoomCR)
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "RPKM")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, RPKM, colour = Cytogenetic_risk)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
```

Ok something is really up. Diffential expression is being called on transcripts with very small RPKM values. Look at the y axis: these values are NOT log transformed! This is very concerning.

I believe I am not analysing the RPKM values correctly during differential expression analysis with voom. 

I will complete differential expression analysis using read count data instead.
