<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>Boxplotter for Differential expression analysis for trisomy 8, deletion 5 and deletion 7</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: rgb(88, 72, 246)
   }

   pre .number {
     color: rgb(0, 0, 205);
   }

   pre .comment {
     color: rgb(76, 136, 107);
   }

   pre .keyword {
     color: rgb(0, 0, 255);
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: rgb(3, 106, 7);
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>




</head>

<body>
<h1>Boxplotter for Differential expression analysis for trisomy 8, deletion 5 and deletion 7</h1>

<blockquote>
<p>This code is for plotting hits of interest from the DEA of the 3key deletion genotypes (trisomy 8, del 5 and del 7).
It performs differential expression analysis on cleaned RNA-seq read count data, and when given abrreviated hit names, 
it finds the full names, prepares the appropriate data matrix for plotting and then boxplots them.</p>
</blockquote>

<h2>Load data and required libraries</h2>

<p>Load RNA-seq data and the experimental design files:</p>

<pre><code class="r">rDes &lt;- read.table(&quot;../data/experimental_design_cleaned.txt&quot;, sep = &quot;\t&quot;, header = TRUE, 
    row.names = 1)  #might need to fix pathname
rDat &lt;- read.table(&quot;../data/aml.rnaseq.gaf2.0_read_count_cleaned.txt&quot;, sep = &quot;\t&quot;, 
    header = TRUE, row.names = 1)  #might need to fix pathname
rDat &lt;- read.table(&quot;../data/aml.rnaseq.gaf2.0_read_count_cleaned.txt&quot;, sep = &quot;\t&quot;, 
    header = TRUE, row.names = 1)  #might need to fix pathname
</code></pre>

<p>Load required libraries:</p>

<pre><code class="r">library(ggplot2)  # we&#39;ll make figures with both
library(reshape2)  # for the function melt
library(limma)
library(edgeR)
# library(RColorBrewer)
</code></pre>

<p>Both the design matrix and the data matrix have already been cleaned, filtered and inspected.</p>

<p>Apply scale normalization:</p>

<pre><code class="r"># normFactor &lt;- calcNormFactors(rDat)
</code></pre>

<p>Preparing model matrix:</p>

<pre><code class="r"># Preparing Model matrices for Voom normalization-------------
modelMat_noInt &lt;- model.matrix(~trisomy_8 * del_5 * del_7, rDes)
# The intercept represents a group with all samples without deletion of
# interest. this group changes depending on the deletion, since some samples
# share multiple deletions.
head(modelMat_noInt)
</code></pre>

<pre><code>##      (Intercept) trisomy_8TRUE del_5TRUE del_7TRUE trisomy_8TRUE:del_5TRUE
## 2803           1             0         0         0                       0
## 2805           1             0         0         0                       0
## 2806           1             0         0         0                       0
## 2807           1             0         0         0                       0
## 2808           1             0         0         0                       0
## 2810           1             0         0         0                       0
##      trisomy_8TRUE:del_7TRUE del_5TRUE:del_7TRUE
## 2803                       0                   0
## 2805                       0                   0
## 2806                       0                   0
## 2807                       0                   0
## 2808                       0                   0
## 2810                       0                   0
##      trisomy_8TRUE:del_5TRUE:del_7TRUE
## 2803                                 0
## 2805                                 0
## 2806                                 0
## 2807                                 0
## 2808                                 0
## 2810                                 0
</code></pre>

<p>Finding genes differentially expressed between deletion type:</p>

<pre><code class="r">dat.voomed &lt;- voom(rDat, modelMat_noInt, plot = TRUE, lib.size = colSums(rDat) * 
    normFactor)
</code></pre>

<pre><code>## Error: object &#39;normFactor&#39; not found
</code></pre>

<pre><code class="r">fit_lm &lt;- lmFit(dat.voomed, modelMat_noInt)
</code></pre>

<pre><code>## Error: object &#39;dat.voomed&#39; not found
</code></pre>

<pre><code class="r">fit &lt;- eBayes(fit_lm)
</code></pre>

<pre><code>## Error: object &#39;fit_lm&#39; not found
</code></pre>

<p>Making topTables for the hits (with no interaction)</p>

<pre><code class="r">ttfit_tris8 &lt;- topTable(fit, number = Inf, coef = &quot;trisomy_8TRUE&quot;, p.value = 1e-05)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">ttfit_del5 &lt;- topTable(fit, number = Inf, coef = &quot;del_5TRUE&quot;, p.value = 1e-05)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">ttfit_del7 &lt;- topTable(fit, number = Inf, coef = &quot;del_7TRUE&quot;, p.value = 1e-05)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<p>there are:</p>

<pre><code>
Error in nrow(ttfit_tris8) : object &#39;ttfit_tris8&#39; not found

</code></pre>

<p>hits for trisomy 8, </p>

<pre><code>
Error in nrow(ttfit_del5) : object &#39;ttfit_del5&#39; not found

</code></pre>

<p>hits for del 5, and </p>

<pre><code>
Error in nrow(ttfit_del7) : object &#39;ttfit_del7&#39; not found

</code></pre>

<p>hits for del 7.</p>

<p><em>Retrieving boxplots for the 4 hits of trisomy 8</em></p>

<pre><code class="r">(ttfit_tris8)
</code></pre>

<pre><code>## Error: object &#39;ttfit_tris8&#39; not found
</code></pre>

<pre><code class="r">
voomTris8genes &lt;- rownames(ttfit_tris8)
</code></pre>

<pre><code>## Error: object &#39;ttfit_tris8&#39; not found
</code></pre>

<pre><code class="r">tris8Dat &lt;- rDat[voomTris8genes, ]
</code></pre>

<pre><code>## Error: object &#39;voomTris8genes&#39; not found
</code></pre>

<pre><code class="r">tris8Dat$Transcript &lt;- rownames(tris8Dat)  #creating a transcript group
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<pre><code class="r">tris8Dat &lt;- melt(tris8Dat, id.vars = &quot;Transcript&quot;, variable.name = &quot;TCGA_patient_id&quot;, 
    value.name = &quot;RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<pre><code class="r">
# cleaup molten RPKM data
tris8Dat$TCGA_patient_id &lt;- gsub(&quot;X&quot;, &quot;&quot;, tris8Dat$TCGA_patient_id)
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<pre><code class="r">tris8Dat$Transcript &lt;- gsub(&quot;[|].*$&quot;, &quot;&quot;, tris8Dat$Transcript)
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<pre><code class="r">
# preparing a small design matrix and merging
miniDes &lt;- rDes[, c(&quot;trisomy_8&quot;, &quot;del_5&quot;, &quot;del_7&quot;)]
miniDes$TCGA_patient_id &lt;- rownames(rDes)
tris8Dat &lt;- merge(tris8Dat, miniDes, by = &quot;TCGA_patient_id&quot;)  #merging
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<p>Boxplot for trisomy 8 genes</p>

<pre><code class="r"># plotting
ggplot(tris8Dat, aes(Transcript, log2(RPKM), colour = trisomy_8)) + geom_boxplot() + 
    facet_wrap(~Transcript, scales = &quot;free&quot;) + theme(axis.text.x = element_blank(), 
    axis.ticks.x = element_blank(), aspect.ratio = 2) + ggtitle(&quot;Hits for Trisomy 8&quot;)
</code></pre>

<pre><code>## Error: object &#39;tris8Dat&#39; not found
</code></pre>

<p><em>Retrieving hit genes from del_5.</em></p>

<p>Getting full names of genes for plotting (note they all use caps).</p>

<pre><code class="r">(ttfit_del5)
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r"># higher
a &lt;- grep(&quot;COL5A1&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">b &lt;- grep(&quot;CEACAM1&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">c &lt;- grep(&quot;CREB3L3&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">
# lower
d &lt;- grep(&quot;RBM22&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">e &lt;- grep(&quot;KDM3B&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">f &lt;- grep(&quot;HARS&quot;, rownames(ttfit_del5))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<pre><code class="r">
# alltogether
(del5_index &lt;- c(a, b, c, d, e, f))
</code></pre>

<pre><code>## Error: object &#39;a&#39; not found
</code></pre>

<pre><code class="r">
# list of rownames for plugging back into the system
voomDel5genes &lt;- rownames(ttfit_del5[del5_index, ])
</code></pre>

<pre><code>## Error: object &#39;ttfit_del5&#39; not found
</code></pre>

<p>This is just the plumbing for graph-making.</p>

<pre><code class="r">del5Dat &lt;- rDat[voomDel5genes, ]  #subsetting transcripts of interest from rDat
</code></pre>

<pre><code>## Error: object &#39;voomDel5genes&#39; not found
</code></pre>

<pre><code class="r">del5Dat$Transcript &lt;- rownames(del5Dat)  #creating a transcript group
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<pre><code class="r">del5Dat &lt;- melt(del5Dat, id.vars = &quot;Transcript&quot;, variable.name = &quot;TCGA_patient_id&quot;, 
    value.name = &quot;RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<pre><code class="r">
# cleaup molten RPKM data
del5Dat$TCGA_patient_id &lt;- gsub(&quot;X&quot;, &quot;&quot;, del5Dat$TCGA_patient_id)  #removing the &#39;X&#39; preceding patient ID names
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<pre><code class="r">del5Dat$Transcript &lt;- gsub(&quot;[|].*$&quot;, &quot;&quot;, del5Dat$Transcript)  #shortening transcript names
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<pre><code class="r">
# miniDes already made while exploring trisomy 8.
del5Dat &lt;- merge(del5Dat, miniDes, by = &quot;TCGA_patient_id&quot;)  #merging
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<p>This is for the actual plot.</p>

<pre><code class="r">ggplot(del5Dat, aes(Transcript, log2(RPKM), colour = del_5)) + geom_boxplot() + 
    facet_wrap(~Transcript, scales = &quot;free&quot;) + theme(axis.text.x = element_blank(), 
    axis.ticks.x = element_blank(), aspect.ratio = 2) + ggtitle(&quot;Hits for Del 5&quot;)
</code></pre>

<pre><code>## Error: object &#39;del5Dat&#39; not found
</code></pre>

<p><em>Exploring hit genes from del_7.</em>
TopTable for genes from del_7.
There are </p>

<pre><code>
Error in nrow(ttfit_del7) : object &#39;ttfit_del7&#39; not found

</code></pre>

<p>, so I will explore the head an tail.</p>

<pre><code class="r">head(ttfit_del7)
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">nrow(ttfit_del7)
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">
# higher
a &lt;- grep(&quot;CDCP1&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">b &lt;- grep(&quot;FAM169A&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">c &lt;- grep(&quot;BEND4&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">
# lower
d &lt;- grep(&quot;LUC7L2&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">e &lt;- grep(&quot;MKRN1&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">f &lt;- grep(&quot;SLC25A13&quot;, rownames(ttfit_del7))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">
# alltogether
(del7_index &lt;- c(a, b, c, d, e, f))
</code></pre>

<pre><code>## Error: object &#39;a&#39; not found
</code></pre>

<pre><code class="r">
# list of rownames for plugging back into the system
voomDel7genes &lt;- rownames(ttfit_del7[del7_index, ])
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<p>This is just the plumbing for graph-making.</p>

<pre><code class="r">del7Dat &lt;- rDat[voomDel7genes, ]  #subsetting transcripts of interest from rDat
</code></pre>

<pre><code>## Error: object &#39;voomDel7genes&#39; not found
</code></pre>

<pre><code class="r">del7Dat$Transcript &lt;- rownames(del7Dat)  #creating a transcript group
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<pre><code class="r">del7Dat &lt;- melt(del7Dat, id.vars = &quot;Transcript&quot;, variable.name = &quot;TCGA_patient_id&quot;, 
    value.name = &quot;RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<pre><code class="r">
# cleaup molten RPKM data
del7Dat$TCGA_patient_id &lt;- gsub(&quot;X&quot;, &quot;&quot;, del7Dat$TCGA_patient_id)  #removing the &#39;X&#39; preceding patient ID names
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<pre><code class="r">del7Dat$Transcript &lt;- gsub(&quot;[|].*$&quot;, &quot;&quot;, del7Dat$Transcript)  #shortening transcript names
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<pre><code class="r">
# miniDes already made while exploring trisomy 8.
del7Dat &lt;- merge(del7Dat, miniDes, by = &quot;TCGA_patient_id&quot;)  #merging
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<pre><code class="r">ggplot(del7Dat, aes(Transcript, log2(RPKM), colour = del_7)) + geom_boxplot() + 
    facet_wrap(~Transcript, scales = &quot;free&quot;) + theme(axis.text.x = element_blank(), 
    axis.ticks.x = element_blank(), aspect.ratio = 2) + ggtitle(&quot;Hits for Del 7&quot;)
</code></pre>

<pre><code>## Error: object &#39;del7Dat&#39; not found
</code></pre>

<p><em>Investigating hit shared by del_5 and del_7</em>
adj. P. Values hit shared by del_5 and del_7</p>

<pre><code class="r"># Retrieving full hit name
(SharedGene &lt;- intersect(rownames(ttfit_del5), rownames(ttfit_del7)))
</code></pre>

<pre><code>## Error: object &#39;ttfit_del7&#39; not found
</code></pre>

<pre><code class="r">
# preparing hit for boxplot
sgDat &lt;- rDat[SharedGene, ]  #subsetting transcripts of interest from rDat
</code></pre>

<pre><code>## Error: object &#39;SharedGene&#39; not found
</code></pre>

<pre><code class="r">sgDat$Transcript &lt;- rownames(sgDat)  #creating a transcript group
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">sgDat &lt;- melt(sgDat, id.vars = &quot;Transcript&quot;, variable.name = &quot;TCGA_patient_id&quot;, 
    value.name = &quot;RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">
# cleaup molten RPKM data
sgDat$TCGA_patient_id &lt;- gsub(&quot;X&quot;, &quot;&quot;, sgDat$TCGA_patient_id)  #removing the &#39;X&#39; preceding patient ID names
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">sgDat$Transcript &lt;- gsub(&quot;[|].*$&quot;, &quot;&quot;, sgDat$Transcript)  #shortening transcript names
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">
# miniDes already made while exploring trisomy 8.
sgDat &lt;- merge(sgDat, miniDes, by = &quot;TCGA_patient_id&quot;)  #merging
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">ggplot(sgDat, aes(Transcript, log2(RPKM), colour = del_5)) + geom_boxplot() + 
    facet_wrap(~Transcript, scales = &quot;free&quot;) + theme(axis.text.x = element_blank(), 
    axis.ticks.x = element_blank(), aspect.ratio = 2) + ggtitle(&quot;Collagen, type V, alpha 1: del 5 grouping&quot;)
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

<pre><code class="r">
ggplot(sgDat, aes(Transcript, log2(RPKM), colour = del_7)) + geom_boxplot() + 
    facet_wrap(~Transcript, scales = &quot;free&quot;) + theme(axis.text.x = element_blank(), 
    axis.ticks.x = element_blank(), aspect.ratio = 2) + ggtitle(&quot;Collagen, type V, alpha 1: del 7 grouping&quot;)
</code></pre>

<pre><code>## Error: object &#39;sgDat&#39; not found
</code></pre>

</body>

</html>

