Random forest exploratory analysis
========================================================

Load the data and metadata. In this analysis, I will be using the cleaned RPKM data.

Load libraries:
```{r}
library(kernlab)
library(cvTools)
library(edgeR)
library(caret)
library(plyr)
library(limma)
library(VennDiagram)
```

```{r}
rDes <- read.delim("../../data/experimental_design_cleaned.txt")
rownames(rDes) <- rDes$TCGA_patient_id
rDat <- read.delim("../../data/aml.rnaseq.gaf2.0_rpkm_cleaned.txt", row.names = 1, check.names = FALSE)
```


## 1) Functions for cross-validation
```{r}
# Function to select features using linear models  
# input.dat: training data set  
# input.labels: true outcomes for training data  
fs.lm <- function(input.dat, input.labels) {
  norm.factor <- calcNormFactors(input.dat)
  design <- model.matrix(~input.labels)
  colnames(design) <- c("Intercept", "Label")
  dat.voomed <- voom(input.dat, design, lib.size = colSums(input.dat) * norm.factor)
  fit <- lmFit(dat.voomed, design)
  ebFit <- eBayes(fit)
  hits <- topTable(ebFit, n = Inf, coef = "Label")
#  train.features <- hits$ID[1:25]
  train.features <- rownames(hits)[1:25]
  return(train.features)
}
```


```{r}
# Function to choose features using correlations to outcomes  
# input.dat: training data set  
# input.labels: true outcomes for training data  
fs.corr <- function(input.dat, input.levels) {
  gene.corrs <- apply(input.dat, 1, function(x) return(suppressWarnings(cor(x, as.numeric(input.levels), method = "spearman"))))
  gene.corrs <- gene.corrs[order(abs(gene.corrs), na.last = NA, decreasing = TRUE)]
  return(names(gene.corrs[1:25]))
}
```

```{r}
# Function to run k-fold cross validation with a random forest  
# all.dat: all data used in the analysis  
# all.labels: true outcomes for the data  
# K: number of folds to use in CV  
# fs.method: the strategy to use for feature selection  
svm.cv <- function(all.dat, all.labels, all.levels, K = 5, fs.method = "lm") {
  set.seed(540)
  folds <- cvFolds(ncol(all.dat), K = K)

  conf_matrix <- matrix(0, nrow = 2, ncol = 2, dimnames = list(c("true0", "true1"), c("pred0", "pred1")))
  feature.list <- list()
  
  for (f in 1:K) {
    train.samples <- folds$subsets[folds$which != f,]
    test.samples <- folds$subsets[folds$which == f,]
   
    if (fs.method == "lm") {
      train.features <- fs.lm(
        all.dat[,train.samples],
        all.labels[train.samples]
      )
    }
    else if (fs.method == "corr") {
      train.features <- fs.corr(
        all.dat[,train.samples],
        all.levels[train.samples]
      )
    }
    feature.list[[paste("Fold", f, sep = "")]] <- train.features
    
    train.dat <- data.frame(
      class = all.labels[train.samples],
      t(all.dat[train.features, train.samples])
    )
    test.dat <- data.frame(
      t(all.dat[train.features, test.samples])
    )
    test.labels <- all.labels[test.samples]
    
    fit.svm <- ksvm(class ~ ., train.dat)
    
    pred.svm <- predict(fit.svm, newdata = test.dat, type = "response")
      
    results <- table(
      factor(test.labels, levels = c(0,1)),
      factor(pred.svm, levels = c(0,1)),
      dnn = c("obs", "pred")
    )
    
    conf_matrix[1,1] <- conf_matrix[1,1] + results[1,1]
    conf_matrix[1,2] <- conf_matrix[1,2] + results[1,2]
    conf_matrix[2,1] <- conf_matrix[2,1] + results[2,1]
    conf_matrix[2,2] <- conf_matrix[2,2] + results[2,2] 
  }
  
  svm.sens <- conf_matrix[2,2] / sum(conf_matrix[2,])
  svm.spec <- conf_matrix[1,1] / sum(conf_matrix[1,])
  svm.acc <- (conf_matrix[1,1] + conf_matrix[2,2]) / sum(conf_matrix)
  
  return(list(acc = svm.acc, sens = svm.sens, spec = svm.spec, feature.list = feature.list))
}
```


## 2) Train svm to predict cytogenetic risk, use linear models for feature selection

Set up the data. Remove samples where the cytogenetic risk category is not determined, and set categories to poor and not poor:
```{r}
svmDes <- rDes[rDes$Cytogenetic_risk != "N.D.",]
svm.labels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(FALSE,FALSE,TRUE), warn_missing = TRUE)
svm.labels <- factor(svm.labels)
svm.levels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(3,2,1), warn_missing = TRUE)
svm.levels <- factor(svm.levels)
svmDat <- rDat[,rownames(svmDes)]
```

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "lm")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```


## 3) Predict the different cytogenetic mutations, use linear models for feature selection

First look at trisomy 8:
```{r}
cv.trisomy8.res <- svm.cv(rDat, factor(as.numeric(rDes$trisomy_8)), svm.levels, K = 5, fs.method = "lm")
cv.trisomy8.res[1:3]
fts <- cv.trisomy8.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.trisomy8 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Next look at deletions of chromosome 5:
```{r}
cv.del5.res <- svm.cv(rDat, factor(as.numeric(rDes$del_5)), svm.levels, K = 5, fs.method = "lm")
cv.del5.res[1:3]
fts <- cv.del5.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.del5 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Next look at deletions of chromosome 7:
```{r}
cv.del7.res <- svm.cv(rDat, factor(as.numeric(rDes$del_7)), svm.levels, K = 5, fs.method = "lm")
cv.del7.res[1:3]
fts <- cv.del7.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.del7 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Now get a sense of how many genes are shared between these 4 different sets:
```{r}
fts.all <- list(risk = common.fts.risk, trisomy8 = common.fts.trisomy8, del5 = common.fts.del5, del7 = common.fts.del7)
plot.new()
venn.plot <- venn.diagram(fts.all, filename = NULL, fill = c("red", "yellow", "blue", "green"))
grid.draw(venn.plot)
```


## 4) Train svm to predict cytogenetic risk, use correlations for feature selection

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "corr")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```


## 5) Predict the different cytogenetic mutations, use correlations for feature selection

First look at trisomy 8:
```{r}
cv.trisomy8.res <- svm.cv(rDat, factor(as.numeric(rDes$trisomy_8)), factor(as.numeric(rDes$trisomy_8)), K = 5, fs.method = "corr")
cv.trisomy8.res[1:3]
fts <- cv.trisomy8.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.trisomy8 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Next look at deletions of chromosome 5:
```{r}
cv.del5.res <- svm.cv(rDat, factor(as.numeric(rDes$del_5)), factor(as.numeric(rDes$del_5)), K = 5, fs.method = "corr")
cv.del5.res[1:3]
fts <- cv.del5.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.del5 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Next look at deletions of chromosome 7:
```{r}
cv.del7.res <- svm.cv(rDat, factor(as.numeric(rDes$del_7)), factor(as.numeric(rDes$del_7)), K = 5, fs.method = "corr")
cv.del7.res[1:3]
fts <- cv.del7.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.del7 <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

Now get a sense of how many genes are shared between these 4 different sets:
```{r}
fts.all <- list(risk = common.fts.risk, trisomy8 = common.fts.trisomy8, del5 = common.fts.del5, del7 = common.fts.del7)
plot.new()
venn.plot <- venn.diagram(fts.all, filename = NULL, fill = c("red", "yellow", "blue", "green"))
grid.draw(venn.plot)
```

## 6) Train svm to predict "Good" cytogenetic risk, use linear models for feature selection

Set up the data. Remove samples where the cytogenetic risk category is not determined, and set categories to good and not good:
```{r}
svmDes <- rDes[rDes$Cytogenetic_risk != "N.D.",]
svm.labels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(1,0,0), warn_missing = TRUE)
svm.labels <- factor(svm.labels)
svm.levels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(3,2,1), warn_missing = TRUE)
svm.levels <- factor(svm.levels)
svmDat <- rDat[,rownames(svmDes)]
```

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "lm")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

## 7) Train svm to predict "Good" cytogenetic risk, using correlations for feature selection

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "corr")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

## 8) Train svm to predict "Intermediate" cytogenetic risk, use linear models for feature selection

Set up the data. Remove samples where the cytogenetic risk category is not determined, and set categories to intermediate and not intermediate:
```{r}
svmDes <- rDes[rDes$Cytogenetic_risk != "N.D.",]
svm.labels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(0,1,0), warn_missing = TRUE)
svm.labels <- factor(svm.labels)
svm.levels <- mapvalues(svmDes$Cytogenetic_risk, c("Good","Intermediate","Poor"), c(3,2,1), warn_missing = TRUE)
svm.levels <- factor(svm.levels)
svmDat <- rDat[,rownames(svmDes)]
```

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "lm")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```

## 9) Train svm to predict "Intermediate" cytogenetic risk, using correlations for feature selection

Run a 5-fold cross-validation for the data:
```{r}
cv.risk.res <- svm.cv(svmDat, svm.labels, svm.levels, K = 5, fs.method = "corr")
cv.risk.res[1:3]
fts <- cv.risk.res[[4]]

plot.new()
venn.plot <- venn.diagram(fts, filename = NULL, fill = c("red", "blue", "green", "yellow", "purple"), margin = 0.1)
grid.draw(venn.plot)

(common.fts.risk <- intersect(fts[[1]], intersect(fts[[2]], intersect(fts[[3]], intersect(fts[[4]], fts[[5]])))))
```